### ISSUE TITLE
Add option for space-based indentation in `ley new` files, make default?

### ISSUE THREAD

**Issue by karlhorky** [lukeed/ley#16]
Hi @lukeed 👋 Hope you are well and everything is going great.

Would it be possible to have an option for `ley new ...` commands that allow setting the indentation to space- or tab-based?

Also, since the common JS style is space-based indentation, would you also consider making this the default? If you would agree here, the option name could also be `--use-tabs`, analogous to [the Prettier option](https://prettier.io/docs/en/options.html#tabs)

**Comment by lukeed** [comment]
Hey, thank you, but things have been better to be perfectly honest 😬 

You should already be using `.editorconfig` files & IDE plugins. It will auto-convert all files (by extension match) to your specified settings. So as soon as you write & save your migration contents, any whitespace will be converted to your preferences.

That said, I won't add a CLI flag for this. It's not a `ley` concern, and there are better tools/solutions out there for this.
Also, for the record, `common JS style is space-based indentation` – this is completely not true. Whitespace has absolutely nothing to do with the module format.. you just may be seeing space-based indentation more often than not.

FWIW, tabs are more accessible and team-friendly. A space is always 1-character width. Tabs are configurable and can have local overrides w/o affecting team members' preferences (see `.editorconfig` once again).

Sorry, but thank you for the suggestion!

**Comment by karlhorky** [comment]
> You should already be using `.editorconfig` files & IDE plugins. It will auto-convert all files (by extension match) to your specified settings. So as soon as you write & save your migration contents, any whitespace will be converted to your preferences.

Hm, I just tried this out, and I couldn't get [the VS Code EditorConfig extension](https://marketplace.visualstudio.com/items?itemName=EditorConfig.EditorConfig) to do what you're suggesting it can. I used an `.editorconfig` file with the following content. There was no auto-formatting:

```ini
[*.js]
indent_style = space
indent_size = 2
```

---

I do have Prettier as well, which gets the user part of the way, but it (or maybe VS Code?) has a weird problem with it:

1. Create a new migration file (this uses tabs)
2. Save the file (tabs converted to spaces by Prettier 🙌)
3. Change the `// <insert magic here>` to ``` await client`` ```, enter into the backticks, and then hit enter
4. Tabs are back (inside the template string)! 😫

I think it may have something to do with the auto-detection of the indentation when the file loads.

---

> Also, for the record, `common JS style is space-based indentation` – this is completely not true. Whitespace has absolutely nothing to do with the module format.. you just may be seeing space-based indentation more often than not.

Ah, I think my words were confusing here. I didn't mean anything to do with the module format. I meant that the most common style in JS code is space indentation.

But no problem, I don't mean to start a religious war about this.

---

I have an alternative that I'd like to propose: just avoid adding indentation. Here's a PR: https://github.com/lukeed/ley/pull/17
